app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -164,9 +191,36 @@ public void deleteAllSubpopulations(StudyIdentifier studyId) {
         if (!subpops.isEmpty()) {
             for (Subpopulation subpop : subpops) {
                 studyConsentDao.deleteAllConsents(subpop.getGuid());
+                criteriaDao.deleteCriteria(subpop.getKey());
             }
             List<FailedBatch> failures = mapper.batchDelete(subpops);
             BridgeUtils.ifFailuresThrowException(failures);
         }
     }
+    
+    private String getKey(Subpopulation subpop) {
+        return "subpopulation:" + subpop.getGuidString();
+    }
+    
+    // Save the criteria object if it exists. If it does not, copy the criteria data from the subpopulation
+    // and save that. Set that on subpopulation. Once all models have a criteria object, this will be removed 
+    // along with the fields on subpopulation.
+    private void persistCriteria(Subpopulation subpop) {
+        Criteria criteria = subpop.getCriteria();
+        Criteria makeCopyOf = (criteria == null) ? subpop : criteria;
+
+        Criteria copy = criteriaDao.copyCriteria(getKey(subpop), makeCopyOf);
+        criteriaDao.createOrUpdateCriteria(copy);
+        subpop.setCriteria(copy);
+    }
+
+    // Load the criteria object. If it doesn't exist, assemble it from the criteria data on the subpopulation
+    // object. Once all models have a criteria object, this will be removed along with the fields on subpopulation.
+    private void loadCriteria(Subpopulation subpop) {
+        Criteria criteria = criteriaDao.getCriteria(getKey(subpop));
+        if (criteria == null) {
+            criteria = criteriaDao.copyCriteria(getKey(subpop), subpop);
+        }
+        subpop.setCriteria(criteria);
+    }
```

Comment:
If as per the previous comment, we make Subpop not implement Criteria, we won't be able to use the copy constructor and may need to duplicate a bit of logic. I think that's fine though, since this is a transitionary period, and that logic will go away.

Alternatively, there are exactly 2 non-trivial Subpopulations in production right now. We could manually migrate those, delete the test subpops, and add some logic such that if Subpop.getCriteria() is null, we trivially match that criteria.